examples/nas/oneshot/spos/search.py

```diff
@@ -37,49 +108,75 @@ def __call__(self, ir_model):
-    transf = [
-        transforms.RandomCrop(32, padding=4),
-        transforms.RandomHorizontalFlip()
-    ]
-    normalize = [
-        transforms.ToTensor(),
-        transforms.Normalize([0.49139968, 0.48215827, 0.44653124], [0.24703233, 0.24348505, 0.26158768])
-    ]
-    # FIXME
-    # CIFAR10 is used here temporarily.
-    # Actually we should load weight from supernet and evaluate on imagenet.
-    train_dataset = serialize(CIFAR10, 'data', train=True, download=True, transform=transforms.Compose(transf + normalize))
-    test_dataset = serialize(CIFAR10, 'data', train=False, transform=transforms.Compose(normalize))
-
-    trainer = pl.Classification(train_dataloader=pl.DataLoader(train_dataset, batch_size=64),
-                                val_dataloaders=pl.DataLoader(test_dataset, batch_size=64),
-                                max_epochs=2, gpus=1)
-
-    simple_strategy = strategy.RegularizedEvolution(model_filter=LatencyFilter(threshold=100, predictor=base_predictor),
-                                                    sample_size=1, population_size=2, cycles=2)
-    exp = RetiariiExperiment(base_model, trainer, strategy=simple_strategy)
+def _main():
+    parser = argparse.ArgumentParser("SPOS Evolutional Search")
+    parser.add_argument("--port", type=int, default=8084)
+    parser.add_argument("--imagenet-dir", type=str, default="./data/imagenet")
+    parser.add_argument("--checkpoint", type=str, default="./data/checkpoint-150000.pth.tar")
+    parser.add_argument("--spos-preprocessing", action="store_true", default=False,
+                        help="When true, image values will range from 0 to 255 and use BGR "
+                             "(as in original repo).")
+    parser.add_argument("--seed", type=int, default=42)
+    parser.add_argument("--workers", type=int, default=6)
+    parser.add_argument("--train-batch-size", type=int, default=128)
+    parser.add_argument("--train-iters", type=int, default=200)
+    parser.add_argument("--test-batch-size", type=int, default=512)
+    parser.add_argument("--log-frequency", type=int, default=10)
+    parser.add_argument("--label-smoothing", type=float, default=0.1)
+    parser.add_argument("--evolution-sample-size", type=int, default=10)
+    parser.add_argument("--evolution-population-size", type=int, default=50)
+    parser.add_argument("--evolution-cycles", type=int, default=10)
+
+    args = parser.parse_args()
+
+    # use a fixed set of image will improve the performance
+    torch.manual_seed(args.seed)
+    torch.cuda.manual_seed_all(args.seed)
+    np.random.seed(args.seed)
+    random.seed(args.seed)
+    torch.backends.cudnn.deterministic = True
+
+    assert torch.cuda.is_available()
+
+    base_model = ShuffleNetV2OneShot()
+    base_latency_predictor = 'cortexA76cpu_tflite21'
+    criterion = CrossEntropyLabelSmooth(1000, args.label_smoothing)
+    if args.spos_preprocessing:
+        trans = nni.trace(transforms.Compose)([
```

Comment:
would also be better to add comment to tell users that why `nni.trace` is used here. to tell users that `nni.trace` is used to make these object serializable, so that the trials can run other other processes or on remote servers.

Reply:
Have fixed it 😃